vsintegration/src/FSharp.Editor/CodeFixes/AddOpenCodeFixProvider.fs

```diff
@@ -50,128 +111,125 @@ type internal AddOpenCodeFixProvider [<ImportingConstructor>] (assemblyContentPr
+
+                        let entities =
+                            assemblyContentProvider.GetAllEntitiesInProjectAndReferencedAssemblies checkResults
+                            |> List.collect (fun s ->
+                                [
+                                    yield s.TopRequireQualifiedAccessParent, s.AutoOpenParent, s.Namespace, s.CleanedIdents
+                                    if isAttribute then
+                                        let lastIdent = s.CleanedIdents.[s.CleanedIdents.Length - 1]
+
+                                        if
+                                            lastIdent.EndsWith "Attribute"
+                                            && s.Kind LookupType.Precise = EntityKind.Attribute
+                                        then
+                                            yield
+                                                s.TopRequireQualifiedAccessParent,
+                                                s.AutoOpenParent,
+                                                s.Namespace,
+                                                s.CleanedIdents
+                                                |> Array.replace (s.CleanedIdents.Length - 1) (lastIdent.Substring(0, lastIdent.Length - 9))
+                                ])
+
+                        ParsedInput.GetLongIdentAt parseResults.ParseTree unresolvedIdentRange.End
+                        |> Option.bind (fun longIdent ->
+                            let maybeUnresolvedIdents =
+                                longIdent
+                                |> List.map (fun ident ->
+                                    {
+                                        Ident = ident.idText
+                                        Resolved = not (ident.idRange = unresolvedIdentRange)
+                                    })
+                                |> List.toArray
+
+                            let insertionPoint =
+                                if document.Project.IsFSharpCodeFixesAlwaysPlaceOpensAtTopLevelEnabled then
+                                    OpenStatementInsertionPoint.TopLevel
+                                else
+                                    OpenStatementInsertionPoint.Nearest
+
+                            let createEntity =
+                                ParsedInput.TryFindInsertionContext
+                                    unresolvedIdentRange.StartLine
+                                    parseResults.ParseTree
+                                    maybeUnresolvedIdents
+                                    insertionPoint
+
+                            entities
+                            |> Seq.map createEntity
+                            |> Seq.concat
+                            |> Seq.toList
+                            |> getSuggestionsAsCodeFixes context sourceText
+                            |> Seq.tryHead))
```

Comment:
Same case as in #15891
The code fix already offers only the first thing, hence also #15911

Will be fixed separately.